src/vllm_router/services/request_service/request.py

```diff
@@ -195,12 +196,19 @@ async def route_general_request(
             logger.warning("Failed to parse rewritten request body as JSON")
 
     # TODO (ApostaC): merge two awaits into one
-    endpoints = get_service_discovery().get_endpoint_info()
+    service_discovery = get_service_discovery()
+    endpoints = service_discovery.get_endpoint_info()
     engine_stats = request.app.state.engine_stats_scraper.get_engine_stats()
     request_stats = request.app.state.request_stats_monitor.get_request_stats(
         time.time()
     )
 
+    aliases = getattr(service_discovery, "aliases", None)
+    if aliases and requested_model in aliases.keys():
+        requested_model = aliases[requested_model]
+        request_body = replace_model_in_request_body(request_json, requested_model)
+        update_content_length(request, request_body)
```

Comment:
This double-lookup into `aliases` could be avoided. WDYT? It is a hot-path that is executed on every request, so this might be worth optimizing although the lookup is not that expensive.

```suggestion
    if aliases:
        if actual_model := aliases.get(requested_model):
            request_body = replace_model_in_request_body(request_json, actual_model)
            update_content_length(request, request_body)
```

Reply:
in my opinion this makes it a bit harder to read, but I will leave it up to the maintainers.

Reply:
The current code looks good to me